src/server/enterprise/server/env.go

```diff
@@ -24,24 +26,65 @@ type Env struct {
 
 	AuthServer    auth.APIServer
 	GetPachClient func(context.Context) *client.APIClient
+	GetKubeClient func() *kube.Clientset
 
 	BackgroundContext context.Context
+	Namespace         string
+	mode              PauseMode
+	UnpausedMode      string
+}
+
+// PauseMode represents whether a server is unpaused, paused, a sidecar or an enterprise server.
+type PauseMode uint8
+
+const (
+	UnpausableMode PauseMode = iota
+	FullMode
+	PausedMode
+)
+
+type Option func(Env) Env
+
+func WithEtcdPrefix(etcdPrefix string) Option {
+	return func(e Env) Env {
+		e.EtcdPrefix = etcdPrefix
+		return e
+	}
 }
 
-func EnvFromServiceEnv(senv serviceenv.ServiceEnv, etcdPrefix string, txEnv *txnenv.TransactionEnv) Env {
-	return Env{
+func WithUnpausedMode(mode string) Option {
+	return func(e Env) Env {
+		e.UnpausedMode = mode
+		return e
+	}
+}
+
+func WithMode(mode PauseMode) Option {
```

Comment:
why do we need mode & unpaused mode? it doesn't look like e.mode gets used right?

Reply:
No, it’s used at ll. 189 & 312, in `Activate` and `Deactivate`, to disable activation & deactivation when in a paused status.

Reply:
ah got it! My b 👍